File: examples/ruby/get_merchant/get_merchant.rb
# frozen_string_literal: true

# Fetch the merchant profile via the public client (Examples deliverable).
# Module-scoped so the name does not collide with Minitest::Test#run.
module GetMerchant
  module_function

  def run(client)
    merchant = client.get_merchant
    raise "get_merchant returned #{merchant.class}, want Hash" unless merchant.is_a?(Hash)

    merchant
  end
end

File: examples/python/get-merchant/get_merchant.py
"""Testable get-merchant example logic (Examples deliverable)."""

from __future__ import annotations

import json
from typing import Any


def run(client: Any) -> dict[str, Any]:
    """Fetch the merchant profile via the low-level client, unwrapping the JSON envelope."""
    envelope = json.loads(client.get_merchant_blocking("{}"))
    if not envelope.get("ok"):
        raise RuntimeError(f"get_merchant failed: {envelope.get('error')}")
    value = envelope.get("value")
    if not isinstance(value, dict):
        raise RuntimeError("get_merchant returned a non-object value")
    return value
